### PR TITLE
Add `close_frame` function to `ws::Message`

### LINKED ISSUE #774: Add function to obtain the close frame information from ws::Message
**Is your feature request related to a problem? Please describe.**
There is no way to obtain the close code or close reason from `ws::Message`. I learnt from this [discussion on discord](https://discord.com/channels/500028886025895936/671042482770935808/796029022281924650) that `Message::into_bytes` returns the close reason but this behavior isn't clear from documentation. Making matters worse, `Message::as_bytes` doesn't have the same behavior, instead [it returns an empty slice](https://github.com/seanmonstar/warp/blob/42fd14fdab8145d27ae770fe4b5c843a99bc2a44/src/filters/ws.rs#L342) when `Message` is of `Close` type.

**Describe the solution you'd like**
A function to obtain the close frame information like: 
```rust
/// (u16, &str) = (close_code, close_reason)
fn close_frame(&self) -> (u16, &str)
```

It would also be nice to have `as_bytes` and `into_bytes` to behave the same way.

**Describe alternatives you've considered**
None available

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by smallglitch**
(Partially) resolves #774

### CODE REVIEW

File: src/filters/ws.rs [hunk 1]
```diff
@@ -324,6 +324,21 @@ impl Message {
         self.inner.is_pong()
     }
 
+    /// Try to get the close frame (close code and reason)
+    pub fn close_frame(&self) -> Result<Option<(u16, &str)>, ()> {
```
Comment (hamza1311) [inline src/filters/ws.rs:328]: Why return a `Result<Option>, ()>` instead of `Option`? `()` doesn't mean anything.

Comment (smallglitch) [reply src/filters/ws.rs:328]: To be consistent with [to_str()](https://github.com/seanmonstar/warp/blob/3bae3ecdad654e3c3145674a6387f2f1214e2ebe/src/filters/ws.rs#L342-L348)

Comment (hamza1311) [reply src/filters/ws.rs:328]: `to_str` returns only a `Result`, not `Result<Option>>`. I don't think that's a good API but I'll leave that to @seanmonstar to decide for sure.

Comment (seanmonstar) [reply src/filters/ws.rs:328]: I don't remember why `to_str` does that... I'm not sure it's the best pattern to follow.

Just an `Option` here seems fine.

Comment (hamza1311) [reply src/filters/ws.rs:328]: Should both of the methods just return `Option` to be consistent? Or is just this one fine?

**Review by seanmonstar — APPROVED** [review]

**Comment by seanmonstar** [comment]
Thanks so much!

**Pull request merged**